python/cudf/cudf/core/frame.py

```diff
@@ -3658,6 +3705,125 @@ def _colwise_binop(
 
         return output
 
+    # For more detail on this function and how it should work, see
+    # https://numpy.org/doc/stable/reference/ufuncs.html
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+        # We don't currently support reduction, accumulation, etc. We also
+        # don't support any special kwargs or higher arity ufuncs than binary.
+        if method != "__call__" or kwargs or ufunc.nin > 2:
+            return NotImplemented
+
+        # Binary operations
+        binary_operations = {
+            # Arithmetic binary operations.
+            "add": "add",
+            "subtract": "sub",
+            "multiply": "mul",
+            "matmul": "matmul",
+            "divide": "truediv",
+            "true_divide": "truediv",
+            "floor_divide": "floordiv",
+            "power": "pow",
+            "float_power": "pow",
+            "remainder": "mod",
+            "mod": "mod",
+            "fmod": "mod",
+            # Bitwise binary operations.
+            "bitwise_and": "and",
+            "bitwise_or": "or",
+            "bitwise_xor": "xor",
+            # Comparison binary operators
+            "greater": "gt",
+            "greater_equal": "ge",
+            "less": "lt",
+            "less_equal": "le",
+            "not_equal": "ne",
+            "equal": "eq",
+        }
+
+        fname = ufunc.__name__
+        if fname in binary_operations:
+            reflect = self is not inputs[0]
+            other = inputs[0] if reflect else inputs[1]
+
+            # These operators need to be mapped to their inverses when
+            # performing a reflected operation because no reflected version of
+            # the operators themselves exist.
+            ops_without_reflection = {
+                "gt": "lt",
+                "ge": "le",
+                "lt": "gt",
+                "le": "ge",
+                # ne and eq are symmetric, so they are their own inverse op
+                "ne": "ne",
+                "eq": "eq",
+            }
+
+            op = binary_operations[fname]
+            if reflect and op in ops_without_reflection:
+                op = ops_without_reflection[op]
+                reflect = False
+            op = f"__{'r' if reflect else ''}{op}__"
+
+            # Float_power returns float irrespective of the input type.
+            if fname == "float_power":
+                return getattr(self, op)(other).astype(float)
+            return getattr(self, op)(other)
+
+        # Special handling for various unary operations.
+        if fname == "negative":
+            return self * -1
+        if fname == "positive":
+            return self.copy(deep=True)
+        if fname == "invert":
+            return ~self
+        if fname == "absolute":
+            return self.abs()
+        if fname == "fabs":
+            return self.abs().astype(np.float64)
```

Comment:
```suggestion
        if fname == "positive":
            return self.copy(deep=True)
        elif fname == "invert":
            return ~self
        elif fname == "absolute":
            return self.abs()
        elif fname == "fabs":
            return self.abs().astype(np.float64)
```

Just a cosmetic nitpick.

Reply:
This is actually a pretty controversial topic :) I believe the majority opinion is in favor of not using elifs after a return since it is implicit in the control flow, as evidenced by the existence of such a check in [clang-tidy](https://clang.llvm.org/extra/clang-tidy/checks/readability-else-after-return.html) (for C++) and [pylint](https://vald-phoenix.github.io/pylint-errors/plerr/errors/refactoring/R1705). [Here's one (opinionated) discussion of the topic](https://dmerej.info/blog/post/else-after-return-yea-or-nay/), and [here's another](https://blog.mozilla.org/nnethercote/2009/08/31/no-else-after-return-considered-harmful/). I would vote to leave this unchanged unless and until we come to some consensus as a team. My 2c: I think there are pros and cons to either approach on a case-by-case basis, and in this case I think it's clearer to think of each potential function name as an early exit before we fall back to the cupy logic.

Reply:
Okay, no issues. Happy to leave it as is. 👍